yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -511,9 +517,9 @@ private[spark] class Client(
      */
     val cachedSecondaryJarLinks = ListBuffer.empty[String]
     List(
-      (args.addJars, LocalResourceType.FILE, true),
-      (args.files, LocalResourceType.FILE, false),
-      (args.archives, LocalResourceType.ARCHIVE, false)
+      (sparkConf.get(JARS_TO_DISTRIBUTE).orNull, LocalResourceType.FILE, true),
```

Comment:
minor: I think you could simplify this a little bit by listing just the options (instead of `sparkConf.get(option).orNull`); if you made the config entries lists defaulting to `Nil` (adding `toSequence` to their builders, and using `withDefault` instead of `optional`) you could save even more code below.